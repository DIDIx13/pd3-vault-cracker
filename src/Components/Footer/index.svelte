<script>
    import GitIcon from '../Icons/Git.svelte';

    let ghCommitHash = '#{GH_COMMIT_HASH}#';
</script>

<style src="./style.scss">

</style>

<div class="background">
    <div class="footer">
        <div class="made_by">
            Made with ðŸ”« by &nbsp;
            <a href="https://github.com/SavageCore" target="_blank">
                SavageCore
            </a>
        </div>
        <div class="version">
            <div class="icon">
                <GitIcon />
            </div>
            <a
                href="https://github.com/SavageCore/pd3-vault-cracker/commit/{ghCommitHash}"
                target="_blank">
                {ghCommitHash.slice(0, 7)}
            </a>
        </div>
    </div>
</div>
