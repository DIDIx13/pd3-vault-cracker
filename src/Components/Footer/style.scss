.footer {
    font-family: "Roboto", sans-serif;
    padding: 0.5rem 1rem;
    color: #fff;
    background-color: #1c1b22;
    width: 100%;
    position: fixed;
    bottom: 0;
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.footer a {
    text-decoration: none;
    color: #fff;
    font-weight: normal;
}

.footer a span {
    font-weight: bold;
}

.made_by {
    display: flex;
    align-items: center;
    margin: 0 auto;
}

.version {
    display: flex;
    align-items: center;
}

.version .icon {
    fill: #f14e32;
    width: 24px;
    margin-right: 0.5rem;
    display: flex;
    align-items: center;
}
