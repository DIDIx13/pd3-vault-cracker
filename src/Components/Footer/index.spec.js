import Footer from './index.svelte';
import { render } from '@testing-library/svelte'

describe('Footer', () => {
    it('renders footer', () => {
        const { container } = render(Footer)

        expect(container).toMatchSnapshot()
    });
});
