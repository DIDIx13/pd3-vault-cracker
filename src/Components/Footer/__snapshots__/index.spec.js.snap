// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders footer 1`] = `
<body>
  <div>
    <div
      class="background"
    >
      <div
        class="footer"
      >
        <div>
          Made with ðŸ”« by
            
          <a
            href="https://github.com/SavageCore"
            target="_blank"
          >
            SavageCore
          </a>
        </div>
      </div>
    </div>
  </div>
</body>
`;
